reconcile/test/test_utils_state.py

```diff
@@ -0,0 +1,65 @@
+import pytest
+import boto3
+from moto import mock_s3
+from reconcile.utils.state import State
+
+
+@pytest.fixture
+def accounts():
+    """Account name is needed to instantiate a State"""
+    return [{'name': 'some-account'}]
+
+
+@pytest.fixture
+def s3_client(monkeypatch):
+    monkeypatch.setenv('AWS_ACCESS_KEY_ID', 'testing')
+    monkeypatch.setenv('AWS_SECRET_ACCESS_KEY', 'testing')
+    monkeypatch.setenv('AWS_SECURITY_TOKEN', 'testing')
+    monkeypatch.setenv('AWS_SESSION_TOKEN', 'testing')
+    monkeypatch.setenv('APP_INTERFACE_STATE_BUCKET', 'some-bucket')
+    monkeypatch.setenv('APP_INTERFACE_STATE_BUCKET_ACCOUNT', 'some-account')
+
+    with mock_s3():
+        s3_client = boto3.client('s3', region_name='us-east-1')
+        yield s3_client
```

Comment:
Any need to `yield` here versus just `return`?

Reply:
Yes, if I just `return` the `s3_client` turns back to the boto3 client and tries to access the real infrastructure.

Reply:
Got it, so I assume that's because with the context manager returning would be out of scope of mock_s3.  Makes sense.